### PR TITLE
Add Base45 (RFC9285) to Multibase table

### PR DESCRIPTION
**Pull request by msporny**
This PR adds support for Base45 encoding ([RFC9285](https://datatracker.ietf.org/doc/html/rfc9285)), which is useful when encoding characters that will be placed into a QR Code. QR Codes have an optimized ALPHANUMERIC format that uses base45 encoding.

### CODE REVIEW

**Comment by davidlehn** [comment]
See also: https://github.com/multiformats/multibase/issues/64

File: multibase.csv
Comment (davidlehn) [inline multibase.csv:20]: Add missing CSV delimiter:
```suggestion
U+0052,     R,          base45,             Base45 RFC9285,                                                 draft
```

**Review by dlongley — APPROVED** [review]

**Review by vmx — APPROVED** [review]
This sounds good for me, though I'd like to get approval also from other folks before merging.

**Comment by ben221199** [comment]
Seems fine to me. However, what about upper/lowercase? If `R` is uppercase, isn't it an idea to register `r` too for lowercase?

**Comment by davidlehn** [comment]
> Seems fine to me. However, what about upper/lowercase? If `R` is uppercase, isn't it an idea to register `r` too for lowercase?

There is a specific alphabet in the RFC. There is no lowercase.  I think `R` was also chosen because it is in that alphabet.

**Comment by davidlehn** [comment]
> I don't see terms like "uppercase", "lowercase", "case-sensitive" or "case-insensitive" in the RFC. The only thing I see is a table with uppercase characters. This doesn't give me a clear indication that base 45 cannot be used with lowercase letters. It only signals some preference.

Section 4.2 says:
> The Alphanumeric mode is defined to use 45 characters as specified in this alphabet.

It then has a table with the characters.  Section 6 says:
> Implementations MUST reject any input that is not a valid encoding. For example, it MUST reject the input (encoded data) if it contains characters outside the base alphabet (in Table 1) when interpreting base-encoded data.

I think that's clear on what alphabet to use for this RFC.  There are certainly other possible base45 alphabets and algorithms, but this RFC seems clear on what to use.  What's the use case for doing otherwise?

**Comment by msporny** [comment]
> I think R was also chosen because it is in that alphabet.

"R" was chosen specifically because lowercase "r" isn't a valid character in Base45, which is largely optimized for use in QR Codes, which means that the entire value we put in the QR Code needs to be Base45-encodable, and lowercase "r" is not. 

We really wanted to use "Q" (for QR Code), but couldn't do so because "Q" was reserved due to early IPFS deployments and backwards-compat reasons.

Hope that helps explain some of the thinking behind the selection.

**Review by rvagg — APPROVED** [review]

**Comment by ben221199** [comment]
Fair enough. I will implement it in my library now it is merged.

@rvagg Can we have test vectors for it?

**Comment by rvagg** [comment]
@ben221199 https://github.com/multiformats/multibase/issues/124

If you get their first, open a PR and ping me and I'll see if I can do a JS implementation. Or if you were to do the JS implementation (js-multiformats) I could do the Go one.

**Pull request merged**
